### DISCUSSION TITLE
Ability to get the "previous" state in the AfterChange Hook

### DISCUSSION THREAD

**Discussion by AlessioGr** [payloadcms/payload#941] [Feature Requests & Ideas]
Having this would be extremely helpful for https://github.com/TimHal/pcms-backpop
Relevant Discord discussion: https://discord.com/channels/967097582721572934/1004009734115966976/1008803750971981834

Say the primary object has a relationship to the secondary object. The secondary object then has that backpopulated field to the primary object.
If you delete the relationship of the primary relationship which points to the secondary relationship, we would need a way to find out the removed reference of that secondary relationship in the AfterChange hook. Otherwise, it becomes very hard to find the object where the backpopulated field would need to be removed (currently that's only possible via iterating and checking all those object, resulting in crap performance).

Hope this makes sense - what do you think?

**Comment by christian-reichart** [comment]
Couldn't you just use the beforeChange hook for that? There you have the data that is going to change and the old state.

**Reply by jmikrut** [reply]
Yes, but I think it would be a trivial add to introduce another arg to the `afterChange` hook called `originalDoc`, like we have elsewhere. This could be a pretty quick PR (like, ~5-10 lines of code changed)!
